### PR TITLE
Make gateway not connected errors consistent

### LINKED ISSUE #2853: Make gateway not connected errors consistent
#### Summary

When caching gateway connection stats on Redis, querying the stats of a not-connected Gateway results in a `redis not found` error, instead of `gateway not connected`

#### Steps to Reproduce

1. Start stack with cache enabled
2. Get connection stats of a gateway that is not connected.

#### What do you see now?

```
$ ttn-lw-cli gtw connection-stats gtw1
error:pkg/redis:not_found (entity not found)
    correlation_id=67ce91f04b7b41f49b78ba991da14bb8
exit status 255
```

#### What do you want to see instead?

```bash
$ cli gtw connection-stats gtw1
error:pkg/gatewayserver:not_connected (gateway `gtw1` not connected)
    gateway_uid=gtw1
    correlation_id=e17381fb1e9c4828b16fd30226465efe
exit status 255
```

#### Environment

3.8.5, with `cache.service=redis`

#### How do you propose to implement this?

Catch the redis not found error

#### How do you propose to test this?

Test locally

#### Can you do this yourself and submit a Pull Request?

yes

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #2853

#### Changes

- Catch Redis not found error, convert to gateway not connected.

#### Testing

Test locally

##### Regressions

There should not be any.

#### Notes for Reviewers

Basing on master, this is fixing an inconsistent error message.

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
- [ ] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/gatewayserver/grpc.go [hunk 1]
```diff
@@ -28,16 +29,19 @@ func (gs *GatewayServer) GetGatewayConnectionStats(ctx context.Context, ids *ttn
 		return nil, err
 	}
 
+	uid := unique.ID(ctx, ids)
 	if gs.statsRegistry != nil {
 		stats, err := gs.statsRegistry.Get(ctx, *ids)
 		if err != nil || stats == nil {
```
Comment (KrishnaIyer) [inline pkg/gatewayserver/grpc.go:35]: Should this be `&&`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fix inconsistent error message responses when retrieving connection stats from GS if the gateway is not connected.
```
Comment (johanstokking) [inline CHANGELOG.md:20]: Remove "Fix"; it's already listed under Fixed

File: pkg/gatewayserver/grpc.go (hunk 1 above)
Comment (johanstokking) [reply pkg/gatewayserver/grpc.go:35]: No, but in case of `nil` I think the `errNotConnected` should be returned. `nil` implies not found here I believe.

Comment (neoaggelos) [reply pkg/gatewayserver/grpc.go:35]: To be honest, the `err != nil` is enough here. We never return `nil` without an error, or an error without a `nil` response. Yet, we do this to be on the safe side and avoid returning a nil response.

**Pull request merged**